tests/unit/local_document_tests.py

```diff
@@ -0,0 +1,388 @@
+import mock
+import json
+import requests
+import os
+
+from cloudant.local_document import LocalDocument
+from cloudant.error import CloudantLocalDocumentException
+
+from .unit_t_db_base import UnitTestDbBase
+
+class CloudantLocalDocumentExceptionTests(unittest.TestCase):
+    """
+    Ensure CloudantLocalDocumentException functions as expected.
+    """
+
+    def test_raise_without_code(self):
+        """
+        Ensure that a default exception/code is used if none is provided.
+        """
+        with self.assertRaises(CloudantLocalDocumentException) as cm:
+            raise CloudantLocalDocumentException()
+        self.assertEqual(cm.exception.status_code, 100)
+
+    def test_raise_using_invalid_code(self):
+        """
+        Ensure that a default exception/code is used if invalid code is provided.
+        """
+        with self.assertRaises(CloudantLocalDocumentException) as cm:
+            raise CloudantLocalDocumentException('foo')
+        self.assertEqual(cm.exception.status_code, 100)
+
+
+class LocalDocumentTests(UnitTestDbBase):
+    """
+    LocalDocument unit tests
+    """
+
+    def setUp(self):
+        """
+        Set up test attributes
+        """
+        super(LocalDocumentTests, self).setUp()
+        self.db_set_up()
+
+    def tearDown(self):
+        """
+        Reset test attributes
+        """
+        self.db_tear_down()
+        super(LocalDocumentTests, self).tearDown()
+
+    def test_constructor(self):
+        """
+        Test instantiating a LocalDocument
+        """
+        doc = LocalDocument(self.db, 'julia006')
+        self.assertIsInstance(doc, LocalDocument)
+        self.assertEqual(doc.r_session, self.db.r_session)
+        self.assertEqual(doc.get('_id'), '_local/julia006')
+
+    def test_constructor_local(self):
+        """
+        Test instantiating a LocalDocument with _local in the docid
+        """
+        doc = LocalDocument(self.db, 'julia006')
+        self.assertIsInstance(doc, LocalDocument)
+        self.assertEqual(doc.r_session, self.db.r_session)
+        self.assertEqual(doc.get('_id'), '_local/julia006')
+
+    def test_document_url(self):
+        """
+        Test that the document url is populated correctly
+        """
+        doc = LocalDocument(self.db, 'julia006')
+        self.assertEqual(
+            doc.document_url, '/'.join(
+                [self.db.database_url, '_local', 'julia006']
+            )
+        )
+
+    def test_document_url_encodes_correctly(self):
+        """
+        Test that the document url is populated and encoded correctly
+        """
+        doc = LocalDocument(self.db, 'http://example.com')
+        self.assertEqual(
+            doc.document_url, '/'.join(
+                [self.db.database_url, '_local', 'http%3A%2F%2Fexample.com']
+            )
+        )
+    
+    @unittest.skipUnless(
+        os.environ.get('RUN_CLOUDANT_TESTS') is not None, 'Skipping Cloudant test'
+    )
+    def test_exists(self):
```

Comment:
What about putting the Cloudant tests into their own class like we have in `database_tests.py`?